src/runtime/vue3/plugin.client.ts

```diff
@@ -0,0 +1,117 @@
+import { globalName, storageKey } from '#color-mode-options'
+import { defineNuxtPlugin } from '#app'
+import { addRouteMiddleware, useRoute } from '#imports'
+import { reactive, watch } from 'vue'
+
+// @ts-ignore
+declare module 'vue-router' {
+  interface RouteMeta {
+    colorMode?: string
+  }
+}
+
+const helper = window[globalName] as unknown as {
+  preference: string
+  value: string
+  getColorScheme: () => string
+  addClass: (className: string) => void
+  removeClass: (className: string) => void
+}
+
+export default defineNuxtPlugin((nuxtApp) => {
+  const colorMode = reactive(nuxtApp.payload.state.colorMode || {
```

Comment:
Shall we use `useState('colorMode', { ...defaults })` instead?

Reply:
The issue is that we want to expose a reactive object rather than a ref.

Reply:
Choosing ref over-reactive object for states was an intentional decision. It would be inconsistent if just choose another style for the module state's style. But even if that's the case, maybe wrap ref by useState to a reactive object can do it?

Reply:
Okay. Should now be as you suggest 👍